dev-docs/source/ReleaseChecklist.rst

```diff
@@ -15,260 +15,327 @@ Roles
 #####
 
 The roles are defined in terms of person responsible.
-This does not mean that the person needs to do the jobs himself/herself, but that they are responsible for ensuring that the work gets done.
-
-* Technical Release Manager - Person responsible for technical tasks such as renaming branches, creating tags, configuring build servers.
-* Release Editor - Person responsible for editing the release notes and giving them a common language, layout, and collecting images.
-* Release Manager - Person in charge of the go/no go decision of the release. The main task is to reiterate the timeline and be collection point for information between all of the Local Project Managers.
-* Local Project Manager(s) - People in charge of communicating with local development teams, facility management, and other people at their sponsoring facility.
-* Quality Assurance Manager - Person responsible for making sure that manual testing has been performed. They will ensure Mantid meets quality requirements before delivery in consultation with the *Release Manager*.
+This does not mean that the person needs to do the jobs himself/herself, but that they 
+are responsible for ensuring that the work gets done.
+
+* :ref:`Local Project Manager(s) <local-project-managers-checklist>` - People in charge 
+  of communicating with local development teams, facility management, and other people 
+  at their sponsoring facility.
+* :ref:`Quality Assurance Manager <quality-assurance-manager-checklist>` - Person responsible 
+  for making sure that manual testing has been performed. They will ensure Mantid meets 
+  quality requirements before delivery in consultation with the *Release Manager*.
+* :ref:`Release Editor <release-editor-checklist>` - Person responsible for editing the 
```

Comment:
Could we slightly tweak the wording here to say that the `Release Editor` is *responsible* for ensuring the release notes are edited etc. In the same way as above really this person is in charge of making sure it is done whether by themselves or someone else.